### ISSUE TITLE
Add metrics reporting mechanism for ESPs

### ISSUE THREAD

**Issue by grassdog** [envato/event_sourcery#31]
Add a mechanism to report metrics from ESPs to different adaptors. e.g. Cloudwatch or Datadog.

Some things worth reporting:
- [ ] Processing Lag
- [ ] Number of events processed per minute
- [ ] Processing time per event?

**Comment by patrobinson** [comment]
I'd love to work on this, unless somebody else plans on it?

**Comment by mjward** [comment]
Go for it @patrobinson ❤️

**Comment by mjward** [comment]
I reckon the Lag Monitors might have come into existence _after_ this issue was created? Sept 2016 the only event sourced app we had was Identity.

**Comment by patrobinson** [comment]
@mjward yep, @stevehodgkiss showed me there are a few implementations of this across Identity and Content. We'll be looking to combine the best of all of them here.
